src/App.tsx

```diff
@@ -1,24 +1,83 @@
-import React from 'react';
+import React, { Props } from 'react';
+import Clock from './components/Clock/Clock';
 import './App.scss';
 
-const App: React.FC = () => {
-  window.setInterval(() => {
-    const date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  isClockVisible: boolean;
+  clockName: number;
 };
 
+class App extends React.Component<Props<State>, State> {
+  state = {
+    isClockVisible: true,
+    clockName: 0,
+  };
+
+  setRandomName = () => {
+    if (this.state.isClockVisible) {
+      this.setState(prevState => {
+        const newName = Math.trunc(Math.random() * 50 + 1);
+
+        // eslint-disable-next-line no-console
+        console.log(`The Clock was renamed from ${prevState.clockName} to ${newName}`);
+
+        return {
+          clockName: newName,
+        };
+      });
+    }
+  };
+
+  showClock = () => {
+    this.setState({ isClockVisible: true });
+  };
+
+  hideClock = () => {
+    this.setState({ isClockVisible: false });
+  };
+
+  render() {
+    const { isClockVisible, clockName } = this.state;
+    const { setRandomName, hideClock, showClock } = this;
```

Comment:
When you have more than 2 constants, will be better to write it in that way:
const {
setRandomName,
hideClock,
showClock,
} = this;